pkg/agent/agentversionsjob.go

```diff
@@ -0,0 +1,207 @@
+package agent
+
+import (
+	"bytes"
+	"encoding/xml"
+
+	"fmt"
+	"io/ioutil"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"net/http"
+
+	"github.com/Axway/agent-sdk/pkg/jobs"
+	"github.com/Axway/agent-sdk/pkg/util/errors"
+	log "github.com/Axway/agent-sdk/pkg/util/log"
+)
+
+var agentURL = map[string]string{
+	"AWSDiscoveryAgent":                      "aws-apigw-discovery-agent",
+	"AWSTraceabilityAgent":                   "aws-apigw-traceability-agent",
+	"AzureDiscoveryAgent":                    "azure-discovery-agent",
+	"AzureTraceabilityAgent":                 "azure-traceability-agent",
+	"EnterpriseEdgeGatewayDiscoveryAgent":    "v7-discovery-agent",
+	"EnterpriseEdgeGatewayTraceabilityAgent": "v7-traceability-agent",
+}
+
+type htmlAnchors struct {
+	VersionList []string `xml:"body>pre>a"`
+}
+
+type version struct {
+	major, minor, patch int
+	val                 string
+}
+
+// agentVersionCheckJob - polls for agent versions
+type agentVersionCheckJob struct {
+	jobs.Job
+	allVersions   []string
+	latestVersion string
+	buildVersion  string
+	dataPlaneType string
+	urlName       string
+}
+
+// Ready -
+func (avj *agentVersionCheckJob) Ready() bool {
+	avj.setURLName()
+	return true
+}
+
+// Status -
+func (avj *agentVersionCheckJob) Status() error {
+	return nil
+}
+
+// Execute - run agent version check job one time
+func (avj *agentVersionCheckJob) Execute() error {
+	if avj.urlName == "AgentSDK" || avj.urlName == "" {
+		err := errors.ErrStartingVersionChecker.FormatError("empty or generic data plane type name")
+		log.Error(err)
```

Comment:
Do we want to hide this error or may be trace it, so that non-Axway agent don't keep having this error

Reply:
I switched to Traces.